code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm

```diff
@@ -843,3 +843,84 @@
 	message_admins("[ADMIN_LOOKUPFLW(obsessed)] has been made Obsessed by the midround ruleset.")
 	log_game("[key_name(obsessed)] was made Obsessed by the midround ruleset.")
 	return TRUE
+
+/// Paradox Clone ruleset
+/datum/dynamic_ruleset/midround/from_ghosts/paradox_clone
+	name = "Paradox Clone"
+	midround_ruleset_style = MIDROUND_RULESET_STYLE_LIGHT
+	antag_datum = /datum/antagonist/paradox_clone
+	antag_flag = ROLE_PARADOX_CLONE
+	enemy_roles = list(
+		JOB_CAPTAIN,
+		JOB_DETECTIVE,
+		JOB_HEAD_OF_SECURITY,
+		JOB_SECURITY_OFFICER,
+	)
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 1
+	weight = 4
+	cost = 3
+	repeatable = TRUE
+	var/list/possible_spawns = list() ///places the antag can spawn
+
+/datum/dynamic_ruleset/midround/from_ghosts/paradox_clone/execute()
+	for(var/turf/warp_point in GLOB.xeno_spawn)
+		if(istype(warp_point.loc, /area/station/maintenance))
+			possible_spawns += warp_point
+	if(!possible_spawns.len)
+		message_admins("No valid spawn locations found for Paradox Clone event, aborting...")
+		return MAP_ERROR
+	return ..()
+
+/datum/dynamic_ruleset/midround/from_ghosts/paradox_clone/generate_ruleset_body(mob/applicant)
+	var/datum/mind/player_mind = new /datum/mind(applicant.key)
+	player_mind.active = TRUE
+
+	var/mob/living/carbon/human/clone = new ((pick(possible_spawns)))
+	player_mind.transfer_to(clone)
+	player_mind.set_assigned_role(SSjob.GetJobType(/datum/job/paradox_clone))
+	player_mind.special_role = ROLE_PARADOX_CLONE
+	player_mind.add_antag_datum(/datum/antagonist/paradox_clone)
+
+	//cloning appearence/name/dna
+	var/datum/antagonist/paradox_clone/cloned = player_mind.has_antag_datum(/datum/antagonist/paradox_clone)
+	var/datum/mind/owner_mind = cloned.original_ref.resolve()
+	var/mob/living/carbon/human/human_cloned = owner_mind.current
+
+	clone.fully_replace_character_name(null, human_cloned.dna.real_name)
+	clone.name = human_cloned.name
+	human_cloned.dna.transfer_identity(clone, transfer_SE=1)
+	clone.age = human_cloned.age
+	clone.underwear = human_cloned.underwear
+	clone.undershirt = human_cloned.undershirt
+	clone.socks = human_cloned.socks
+	for(var/datum/quirk/target_quirk as anything in human_cloned.quirks)
+		clone.add_quirk(target_quirk.type)
+	clone.updateappearance(mutcolor_update=1)
+	clone.update_body()
+	clone.domutcheck()
+
+	//cloning clothing/ID/bag
+	clone.mind.assigned_role = human_cloned.mind.assigned_role
+
+	if(isplasmaman(human_cloned))
+		clone.equipOutfit(human_cloned.mind.assigned_role.plasmaman_outfit)
+		clone.internal = clone.get_item_for_held_index(1)
+	clone.equipOutfit(human_cloned.mind.assigned_role.outfit)
+
+	var/obj/item/clothing/under/sensor_clothes = clone.w_uniform
+	var/obj/item/modular_computer/pda/messenger = locate() in clone
+	if(messenger)
```

Comment:
Would all this be better if it was on the antagonist itself? You find the target there, you could also make it mimic said target after that as well?